scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -267,6 +266,9 @@ def add_parse_arguments(parser = None) -> argparse.ArgumentParser:
         name (section.subsection) and 'fifo_api_1cpu.fifo_loop' is
         a Ztest suite_name.test_name identificator.
         Example_2: 'debug.coredump.logging_backend' is a standalone test scenario name.
+        Note: This selection mechanism works only for Ztest suite and test function names in
+        the source files which are not generated by macro-substitutions.
+        Note: With --no-detailed-test-id use only Ztest identificator without scenario name prefix.
```

Comment:
That note at the end i do not understand, what is Ztest Identificator? sounds like a new terminology.

I would also reduce verbosity in the help for command line option and prefer to expand in the documentation but this seems to be a common theme across all options, need to clean this up a bit, we have too many options and each comes with a mouthful of help making --help less helpful :(

Anyways, what I noticed is the following:

```
./scripts/twister --test-tree -T tests/kernel/semaphore/ --no-detailed-test-id
ZEPHYR_BASE unset, using "/home/nashif/zephyrproject/zephyr"
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.8
INFO    - Using Ninja..
INFO    - Zephyr version: v4.0.0-1232-gcacd64a67ae1
INFO    - Using 'zephyr' toolchain.
Testsuite
├── Samples
└── Tests
    ├── kernel_sys_sem
    ├── semaphore
    ├── semaphore_1cpu
    └── semaphore_null_case

```

vs

```
./scripts/twister --test-tree -T tests/kernel/semaphore/
ZEPHYR_BASE unset, using "/home/nashif/zephyrproject/zephyr"
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.9
INFO    - Using Ninja..
INFO    - Zephyr version: v4.0.0-1232-gcacd64a67ae1
INFO    - Using 'zephyr' toolchain.
Testsuite
├── Samples
└── Tests
    └── kernel
        └── semaphore
            ├── kernel.semaphore.semaphore.k_sem_correct_count_limit
            ├── kernel.semaphore.semaphore.k_sem_define
            ├── kernel.semaphore.semaphore.k_sem_init
            ├── kernel.semaphore.semaphore.sem_count_get
            ├── kernel.semaphore.semaphore.sem_give_from_isr
            ├── kernel.semaphore.semaphore.sem_give_from_thread
            ├── kernel.semaphore.semaphore.sem_give_take_from_isr
            ├── kernel.semaphore.semaphore.sem_measure_timeout_from_thread
            ├── kernel.semaphore.semaphore.sem_measure_timeouts
            ├── kernel.semaphore.semaphore.sem_multi_take_timeout_diff_sem
            ├── kernel.semaphore.semaphore.sem_multiple_threads_wait
            ├── kernel.semaphore.semaphore.sem_reset
            ├── kernel.semaphore.semaphore.sem_reset_waiting
            ├── kernel.semaphore.semaphore.sem_take_multiple
            ├── kernel.semaphore.semaphore.sem_take_no_wait
            ├── kernel.semaphore.semaphore.sem_take_no_wait_fails
            ├── kernel.semaphore.semaphore.sem_take_timeout
            ├── kernel.semaphore.semaphore.sem_take_timeout_fails
            ├── kernel.semaphore.semaphore.sem_take_timeout_forever
            ├── kernel.semaphore.semaphore.sem_thread2isr
            ├── kernel.semaphore.semaphore.sem_thread2thread
            ├── kernel.semaphore.semaphore_1cpu.sem_multiple_take_and_timeouts
            ├── kernel.semaphore.semaphore_1cpu.sem_queue_mutual_exclusion
            ├── kernel.semaphore.semaphore_1cpu.sem_take_timeout_isr
            ├── kernel.semaphore.semaphore_null_case.sem_count_get_null
            ├── kernel.semaphore.semaphore_null_case.sem_give_null
            ├── kernel.semaphore.semaphore_null_case.sem_init_null
            ├── kernel.semaphore.semaphore_null_case.sem_reset_null
            ├── kernel.semaphore.semaphore_null_case.sem_take_null
            ├── kernel.semaphore.usage.kernel_sys_sem.multi_thread_sem_limit
            └── kernel.semaphore.usage.kernel_sys_sem.multiple_thread_sem_usage
```

So, this is completeltly different output.